### ISSUE TITLE
Token Holders should be able to vote on posts before they get posted by Koru

### ISSUE THREAD

**Issue by hilmarx** [gelatodigital/koru-dao-contracts#3]
**Bounty:**

1 KoruDAO NFT 

**Problem:**

Right now, posts from NFT holders get posted by the DAO right away. This can potentially lead to problems later on if someone accumulates a lot of NFTs and starts spamming the DAO. 

**Proposed Solution: The Post Queue** 

A potential solution to this would be to create a “post queue”, which is like a pending state between when an NFT holder posts something and the DAO account actually minting a post NFT on Lens. 

We could set a **ACCEPTANCE_THRESHOLD** of e.g. 10 votes which posts have to pass before Koru will actually post the transaction on-chain. This way, token holders can remove certain posts that are “spammy” or “bad” before they get posted on Lens by the DAO. This would work by defining an `ACCEPTANCE_DURATION`, e.g. 2hs, which is the time your posts need to accumulate 10 votes. If it doesnt, then it will not get posted. 

This whole "curation process" of course takes time and attention from all token holders. However, in the future we can build a mechanism which rewards "active" token holders which help with the curation and signalling with dao revenue. 

**Acceptance Criteria:** 

- [ ] Written spec which gets approved by @hilmarx about how to implement the "post queue". 
- [ ] Smart contracts implementing the post queue
- [ ] Integration with Gelato Relay to make everything gasless
- [ ] Test suite with unit tests
- [ ] PR to UI which adds a feature where token holders can press thumbs up to signal that a post can be posted (gasless)

**Comment by NelsonRodMar** [comment]
I love this idea, and as mentioned on Lens by 0xjean.lens we could use [Kleros Moderate](https://kleros.io/moderate/) 

src : open.withlens.app/post/0xa738-0xdc

**Comment by pkazo** [comment]
yep that would help resolve issues, especially when people post unethical or spammy stuff. Or maybe integrate some sort of AI(chatGPT alike) context checker, this may speed up duration and approval of content in general.

**Comment by hilmarx** [comment]
problem with chat gpt is that you can also post via the SCs directly without having to use the UI, which means that someone else could just spin up another front end. Thus more a short term fix. 

Regarding Kleros, sounds interesting but I think something simple as a task queue should do the trick.

**Comment by EthWarrior** [comment]
Could be interersting to have a dynamic voting that either adds (or even posts gets enought votes not to pass voting threshold) time or substracts time to post. This would incentivise to reduce the time for interesting posts the community finds. Let say every vote reduces 30 min (can be tweaked). Of course need to think the edge cases to avoid being gamed.

**Comment by hilmarx** [comment]
> Could be interersting to have a dynamic voting that either adds (or even posts gets enought votes not to pass voting threshold) time or substracts time to post. This would incentivise to reduce the time for interesting posts the community finds. Let say every vote reduces 30 min (can be tweaked). Of course need to think the edge cases to avoid being gamed.

Yes very interesting, for simplicity purposes you also just say once the threshold of e.g. 10 net upvotes (minus downvotes) has been reached, that everyone can just release the post. 

This whole "upvoting" vs "downvoting" has to be incentivized at some point, e.g. with the resale revenue of the NFTs. The more you "curate", the more you receive from the DAO via weekly distributions or smth. As this adds quite some complexity I would have this be a future feature though,

**Comment by hilmarx** [comment]
First implementation, need a review: 

https://github.com/gelatodigital/koru-dao-contracts/pull/7
